### PR TITLE
Web sheet full width migrate 90

### PR DESCRIPTION
**Pull request by njeudy**
- Migrate to V9
- Add README.rst with OCA infos
- Test for community and entreprise version.

### CODE REVIEW

File: web_sheet_full_width/README.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Show sheet over full width
```
Comment (pedrobaeza) [inline web_sheet_full_width/README.rst]: Please use latest OCA's template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: web_sheet_full_width/README.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+==============================
```
Comment (pedrobaeza) [inline web_sheet_full_width/README.rst]: This should have the same length of the title

**Comment by pedrobaeza** [comment]
And please squash your commits in one, as there are little changes.

**Comment by pedrobaeza** [comment]
Tested on runbot, it works OK.

File: web_sheet_full_width/__openerp__.py [hunk 3]
```diff
@@ -43,7 +43,7 @@
         "view/qweb.xml",
     ],
     "auto_install": False,
-    'installable': False,
+    'installable': True,
     "application": False,
     "external_dependencies": {
```
Comment (elicoidal) [inline web_sheet_full_width/__openerp__.py]: Please remove empty keys

**Comment by njeudy** [comment]
OK thanks for the review. I updated all I think.

**Comment by lmignon** [comment]
@pedrobaeza @njeudy We should advertise that the module can only be installed on the community edition (the module web in the enterprise edition is a non lgpl fork of the module web in the community edition not an extend)

**Comment by rvalyi** [comment]
+1 @lmignon you are absolutely right

**Comment by pedrobaeza** [comment]
I'm very worried about this thing, because it means that everyone with the enterprise edition cannot install any of the OCA/web modules, isn't it? Cannot we make an enterprise version of the module with a compatible license?

**Comment by rvalyi** [comment]
@pedrobaeza if so I suggest we sell them on the app-store at the profit of the OCA. My 2 cts.

**Comment by rvalyi** [comment]
@pedrobaeza from what we hear, it's not like so many on-premise users choose the Enterprise proprietary version; eventually that offer will simply die just like the certification program or the SAP indexed prices, may be not worth harming OCA licenses for it. And as for Odoo saas users they cannot install community modules anyway so...

**Comment by pedrobaeza** [comment]
Hehe, I knew your answer when I was writing my comment :stuck_out_tongue_winking_eye:, but this is very serious and it's a clear example of incompatibility: as Odoo has decided to release a separate enterprise web module instead of mounting over the community web module, this makes in practical terms incompatible with a lot of OCA modules, and for me this means that Odoo has dug its own grave.

**Comment by rvalyi** [comment]
@pedrobaeza it's not like we never warned them...

**Comment by lmignon** [comment]
@pedrobaeza I don't know how it will be possible to migrate OCA/geospatial to 9.0...

**Comment by pedrobaeza** [comment]
Yeah, I have also read about that, but it can be migrated for Community version, isn't it?

**Comment by lmignon** [comment]
@pedrobaeza yes but not run on enterprise at all.

**Comment by pedrobaeza** [comment]
This is really worrying... The rest of the @OCA/board should be aware of this.

**Comment by rvalyi** [comment]
This only my opinion here. But I really don't believe the current Enterprise offer where partners can install some proprietary modules on-premise is going anywhere. We might be worried or not (I' m not), but I don't believe that deserves changing our OCA licensing. Anyway this outcome was entirely predictable for years now...

**Comment by njeudy** [comment]
@lmignon @rvalyi @pedrobaeza: this module is only css. can you explain me with it could not be installed on entreprise version ? I juste add some css rules to odoo et doesn't change any part of the code...

**Comment by njeudy** [comment]
We have to change licence to LGPL for web modules ?

**Comment by rvalyi** [comment]
@njeudy in this specific case where it's only css, then may be you may re-license it to some more permissive license to make things clear (I wasn't aware it was only css).
But still this is a serious issue for modules extending the web module in general.

**Comment by max3903** [comment]
@njeudy If you want to allow Enterprise users to use your module, change the license of your module to LGPL. Otherwise, use AGPL.

**Comment by rvalyi** [comment]
@max3903 @njeudy @pedrobaeza
in fact I'm not sure LGPL can derivate from something proprietary.
Does somebody have any clue about that?

**Comment by pedrobaeza** [comment]
Seeing the post about licenses: https://odoo-community.org/blog/oca-news-1/post/oca-odoo-meeting-on-licenses-21, we cannot put an AGPL module over the propietary, but we can put a LGPL. The question is if the authors want to relicense the module to LGPL.

**Comment by rvalyi** [comment]
@pedrobaeza it's written by Odoo SA indeed. Now is there any other source about LGPL telling that? It's just that in the past SA has been so approximative with licensing (see proprietary modules sold with AGPL v8...) that I don't really take everything they say for granted...

**Comment by pedrobaeza** [comment]
Yeah, you're right about the uniqueness of the source. Anyway, I can't imagine an scenario where putting as LGPL can cause any trouble. Do you?

**Comment by rvalyi** [comment]
@pedrobaeza I'm sure LGPL can be extended by proprietary, I'm just not sure about the reverse. That's probably possible, but clearing doubts would be great.

**Comment by njeudy** [comment]
but I ported a module developed by THERP .. so I need to ask him for LGPL ?

**Comment by pedrobaeza** [comment]
Yeah. As now, we have the security that it's incompatible with enterprise edition. Converting to LGPL, @rvalyi expresses also his doubts, but there are chances that it is compatible. You have to ask permission to the authors of the module, not the company.

**Comment by lmignon** [comment]
@njeudy Do you know if your module works on entreprise? If it doesn't work, you just need to advertise that the module doesn't work on enterprise and can't be installed on enterprise. No licence change required in this case.

**Comment by rvalyi** [comment]
@lmignon the Enterprise does not work. How can a community module work on it?
ok that one was free bashing.
-------> []

**Comment by njeudy** [comment]
@lmignon yes it works on entreprise :)

**Comment by pedrobaeza** [comment]
Let's query @hbrunn what he thinks about this and if he wants to switch to LGPL.

**Comment by hbrunn** [comment]
I believe in viral licenses for the time being, so no. But wasn't part of the evil license discussion that OCA modules grant an exception for lgpl core modules?

**Comment by guewen** [comment]
@hbrunn the exception states that an OCA AGPLv3 add-on can be installed aside an Enterprise proprietary add-on but not depend on it.

**Comment by hbrunn** [comment]
thanks for the clarification

**Comment by jgrandguillaume** [comment]
Hi,

I confirm what has been stated here :https://odoo-community.org/blog/oca-news-1/post/oca-odoo-meeting-on-licenses-21
We made our best that time to protect the interest of our contributors.

So, in brief:
-  an OCA AGPLv3 add-on can be installed aside an Enterprise proprietary add-on but not depend on it
- of course he contrary is true too (AGPL cannot extend Enterprise proprietary)
- to extend the Enterprise proprietary module with an OCA one, you need LGPL (or MIT eventually)
- to change the license, you need the agreement of all the authors. The OCA will not use its copyright granted through the CLA for this. An exception may be made once, but the assembly would have to vote.

Regards,

Joël

@lmignon For the Geospatial addons, we're working on it. It'll be available on community version.

**Comment by pedrobaeza** [comment]
OK, so let's add for now the sentence "This module can only be installed on the community edition due to license incompatibilities" and we can continue.

**Comment by elicoidal** [comment]
+1

**Comment by dreispt** [comment]
> the module web in the enterprise edition is a non lgpl fork of the module web in the community edition not an extend

This is a mess - someone didn't do their homework (again). They only had to create a `web_enterprise`overriding stuff from the community `web` and all would be good!

As is causes for a lot of OCA modules to be incompatible with Enterprise.
Not only all OCA/web depend on `web`, but also modules in other repos extend on `web`, such as some timesheet modules.

Changing OCA modules licenses won't fix the root of problem. Odoo must fix doing a proper implementation of the enterprise features, that stays true to what they promised to the community.

**Comment by elicoidal** [comment]
I propose that the OCA board prepare an official request to Odoo, sa.
I fear though that they wont commit such an extensive change for v9 and target might be v10 only.
An additional decision is to be made about the attitude of the OCA for v9.

**Comment by hbrunn** [comment]
how about OCA providing an LGPL module that depends on `web` (enterprise version) and is called `web_enterprise`? This way, we have a marker what is targetted to the community version and what to the enterprise version. And if Odoo SA agrees with the rename, we simply remove this module again.

**Comment by hbrunn** [comment]
PS: Shouldn't we move this to the mailinglist instead of hijacking the PR?

**Comment by lmignon** [comment]
@hbrunn :+1:  for the mailinglist

**Comment by elicoidal** [comment]
+1 for the module
+1 for bringing more light to this discussion

File: web_sheet_full_width/README.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+for form view
+
+Installation
+============
+
+To install this module, you need to:
+
+* Just click install button.
+
+Configuration
```
Comment (lmignon) [inline web_sheet_full_width/README.rst]: You can remove this section since  it's useless

File: web_sheet_full_width/README.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+To configure this module, you need to:
+
+* N/A
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Go to ...
```
Comment (lmignon) [inline web_sheet_full_width/README.rst]: These 2 previous sentences can be removed

File: web_sheet_full_width/README.rst [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/162/9.0
+
+.. repo_id is available in https://github.com/OCA/maintainer-tools/blob/master/tools/repos_with_ids.txt
+.. branch is "8.0" for example
```
Comment (lmignon) [inline web_sheet_full_width/README.rst]: These commented lines can be removed.

**Comment by OSguard** [comment]
I think for OCA it is impossible to maintain module that extend enterprise stuff. The reason is the enterprise license itself. We as partner has access to the enterprise code, but not allow to modified or distribute outside the company. So for modules depends on enterprise we can not use travis, runbot and so on to make sure that our OCA module works with the enterprise version.
As it is not possible to have our toolset work with enterprise stuff, we can not assure the OCA quality. So then we should simply don't do it.

**Comment by rvalyi** [comment]
+1 OSguard
Not to say that if the OCA supports these proprietary modules, then we will have them around for more time. I would say the sooner they are discontinued or kicked out of the market, the better for us.

**Comment by pedrobaeza** [comment]
@rvalyi, OCA cannot and shouldn't make boycott of any of the Odoo ways of financing while they still keeps a little commitment with the community. We only must guarantee the rights of our contributors and check for possible illegal actions.

**Comment by rvalyi** [comment]
@pedrobaeza if financing means promising to VC's they'll close the thing and kill the community eco-system subtlety years after years in exchange of their money, then I believe the sooner we get out of this scheme, the saner for everybody. Sorry I hijacked the thread again, I wasn't sure this point would surface again.

**Comment by jgrandguillaume** [comment]
+1 for the web_enterprise module
+1 for bringing this discussion on ML

For runbot and travis, we can ask Odoo sa to grant the OCA bot user an access to the enterprise repo to allow us to make the build on travis and runbot.

**Comment by njeudy** [comment]
Hello,

What do I need to do so ? wait for OCA decision ?
For now All modules depending on `web` in OCA repos can't be migrate to 9.0 ?

If I have entreprise version it's not legal to use this module ?

sorry for my question but I'm not "licensing system master" I don't know if I should continue migrating web module to V9 :)

**Comment by njeudy** [comment]
@lmignon will apply change soon, thanks

**Comment by lmignon** [comment]
@njeudy 

> If I have entreprise version it's not legal to use this module ?

IMO yes it's not legal to use this module in enterprise

**Comment by pedrobaeza** [comment]
@njeudy, can you please attend the comments and I'll merge?

**Comment by njeudy** [comment]
@pedrobaeza : it's done sorry for the delay :)

**Comment by njeudy** [comment]
@pedrobaeza do you know why this coverage is not ok ?

**Comment by pedrobaeza** [comment]
It was probably a problem in the CIs in that moment. I have finally merged amending the README with latest agreement. Thanks for all your effort!

**Pull request merged**